src/pip/_internal/req/req_uninstall.py

```diff
@@ -312,6 +312,7 @@ def __init__(self, dist: BaseDistribution) -> None:
         self._pth: Dict[str, UninstallPthEntries] = {}
         self._dist = dist
         self._moved_paths = StashedUninstallPathSet()
+        normalize_path_cached.cache_clear()
```

Comment:
It seems very weird to me instantiating a new object of this class has a global side effect. The cache should probably live on this object instead of a global function?

Reply:
> It seems very weird to me instantiating a new object of this class has a global side effect. The cache should probably live on this object instead of a global function?

My thinking was that this might be more generally useful, but you raise a good point. I think I'd be fine with that, but could you help me with the syntax on that? I only know how to apply `lru_cache` to a function, not locally. In that case, perhaps we should give `is_local` its own cache?

Reply:
> Since this function is used by three values in this class that we know upfront anyway, maybe it’d be a good idea to make those values `cached_property`?

I don't think that works. https://docs.python.org/3/faq/programming.html#faq-cache-method-calls says:
> The cached_property approach only works with methods that do not take any arguments.

Reply:
Got it figured out.